### PR TITLE
Added nixos-rebuild completer

### PR DESCRIPTION
**Pull request by aftix**
- Added completer for nixos-rebuild
- Fixed nix completer `builders` and `substituters` flags not being StringArray

### CODE REVIEW

File: completers/nixos-rebuild_completer/cmd/root.go [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+}
+
+func init() {
+	carapace.Gen(rootCmd).Standalone()
+
+	rootCmd.PersistentFlags().Bool("help", false, "Show usage information")
+	rootCmd.PersistentFlags().Bool("offline", false, "Disable substituters and consider all previously downloaded files up-to-date")
+	rootCmd.PersistentFlags().StringSlice("option", []string{}, "Set the Nix configuration setting name to value")
+	rootCmd.PersistentFlags().Bool("refresh", false, "Consider all previously downloaded files out-of-date")
+	rootCmd.PersistentFlags().Bool("version", false, "Show version information")
```
Comment (rsteube) [inline completers/nixos-rebuild_completer/cmd/root.go:28]: I think I've got the wrong version of `nixos-rebuild`?
It doesn't have some flags like `--version` and `--access-tokens`.
It seems to accept the others on root level as well (PersistentFlags) :thinking: .

Comment (aftix) [reply completers/nixos-rebuild_completer/cmd/root.go:28]: What version do you have?

Comment (rsteube) [reply completers/nixos-rebuild_completer/cmd/root.go:28]: The ones that gets installed from `23.11` and `unstable`.

Comment (aftix) [reply completers/nixos-rebuild_completer/cmd/root.go:28]: It seems you're right - the man page said the flags needed to come after the subcommand, so I didn't test them before, but everything (except `--json` on `list-generations`) seems to actually be PersistentFlags. The nixos-rebuild man page also claims it accepts the flags from the `nix` command, but apparently it's only a subset

Comment (aftix) [reply completers/nixos-rebuild_completer/cmd/root.go:28]: Testing further, it seems the command accepts flags beyond the list the specifically call out in the man page (for example nixos-rebuild accepts the `--offline` flag from the nix command), but it isn't consistent in that it accepts "Logging-related options" or other option groups, the accepted flags seems to be taken arbitrarily

Comment (rsteube) [reply completers/nixos-rebuild_completer/cmd/root.go:28]: https://github.com/NixOS/nixpkgs/blob/nixos-23.11/pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh#L48

Comment (aftix) [reply completers/nixos-rebuild_completer/cmd/root.go]: I've updated the PR to make the flags persistent and to only accept flags found in that script

**Review by rsteube — APPROVED** [review]

**Pull request merged**